yt/geometry/particle_oct_container.pyx

```diff
@@ -1033,6 +1033,21 @@ cdef class ParticleBitmap:
     def reset_bitmasks(self):
         self.bitmasks._reset()
 
+    def correct_index_order(self, fname):
```

Comment:
I don't understand what's happening here. Since `self.hash_value` is a function of `self.index_order1` and `self.index_order2` there's no way that `grp.attrs["index_order1"], grp.attrs["index_order2"]` read from cache will be different from `self.index_order1` and `self.index_order2`

Reply:
@Xarthisius here's what happens in order (in `main` right now).

These two steps are the same regardless of whether or not the EWAH file already exists:

1. yt sets `index_order1`, `index_order2`
2. yt creates `ParticleBitmap` object as `regions` attribute of `index` (line 162 of `particle_geometry_handler.py`), passes it _initial_ values of `index_order1` and `index_order2` to create `hash_value` in the `ParticleBitmap` object

Now the "a" case is what happens when there is no EWAH file:

3a. If there is no EWAH file yet, yt creates coarse and fine indices, and (sometimes) decides that `index_order2` needs to be updated. 
4a. yt writes EWAH file--with the _updated_ `index_order2` as the underlying, but `hash_value` is not affected (and hence not the HDF5 group name) since it was already set and it is not updated.

The "b" case is what happens if there is an EWAH file:

3b. yt reads the bitmap masks from the HDF5 group, has no problem doing so because yt is using the `hash_value` from initial `index_order1` and `index_order2`.
4b. Because we read the masks from the EWAH file, yt never drops into `_initialize_coarse_index` where the index order is updated, so whenever we do a calculation, the initial `index_order2` is used but the bitmap masks are using the updated one

Very sneaky...

Reply:
> 4a. yt writes EWAH file--with the _updated_ `index_order2` as the underlying, but `hash_value` is not affected (and hence not the HDF5 group name) since it was already set and it is not updated.

That's the bug! It means I messed it up. `hash_value` should always match current `index1` and `index2`, not just whatever was passed to the constructor.

Reply:
Also the notion that we store meaningful value, that's not just some initial guess, in `ds.index_order` is scary. What happens if I select two spheres and want them to use different indices? Is that disallowed?

Reply:
So that means you have to update it I guess before it gets written--but now the problem is that yt when it's searching for the masks in the EWAH file never gets to the place where it realizes that index_order2 should be updated, and thus it has the wrong hash the second time around. So that's still an issue.

Reply:
So this means we have to open the EWAH file before even creating the `regions` object so we can determine the correct `index_order2`, or something like that. Previous to the HDF5 change, this was handled by reading `index_order2` off from the filename, which the current code is not doing at all